lib/oauth/types.go

```diff
@@ -0,0 +1,12 @@
+package oauth
+
+import (
+	"github.com/enfabrica/enkit/lib/khttp/kcookie"
+	"net/http"
+)
+
+type IAuthenticator interface {
```

Comment:
Can you add a coment like:

```
// An IAuthenticator is any object capable of performing authentication for a web server.
// PerformLogin initiates the login process.
// PerformAuth is invoked at the end, to verify that the login was successful.
// Complete is used to verify that the returned AuthData indicates that the process is now complete.
```

Would be good to add an example/move the example on top of oauth.go to be here. Another day.

Reply:
I also thing it would be a good idea to change this to authenticator at some point, and have single.go and multi.go